File: StraightLine.java
package dymetric;

import java.awt.*;

public class StraightLine extends Canvas {

    protected Color ball_colour;
    protected int x1 = 50;
    protected int x2 = 800;
    protected int y1 = 50;
    protected int y2 = 300;
    protected int x = x1;
    protected int y = y1;
    protected double m, c; // slope and y-intercept of a straight line
    protected final int aWIDTH, aHEIGHT;

    public StraightLine() {
        setBackground(Color.LIGHT_GRAY); // canvas color
        ball_colour = Color.RED;
        aWIDTH = aHEIGHT = 10;

        m = (double) (y2 - y1) / (x2 - x1); // slope
        c = (double) (x2 * y1 - x1 * y2) / (x2 - x1); // y-intercept

    }

    // Feel free to double buffer if flickering occurs
    public void paint(Graphics g) {
        g.setColor(ball_colour);
        // draw a dot
        g.fillOval(x, y, aWIDTH, aHEIGHT);
    }

    public void moveInLine() {
        // condition for continuing motion
        while (x <= 700) {
            y = (int) Math.ceil(m * x + c);
            paint(this.getGraphics());
            
            x += 10;
            // introduce a delay betweeen renderings
            try {
                Thread.sleep(50);
            } catch (InterruptedException e) {
            }
        }
    }
}

File: StraightLine.cpp
#include "stdafx.h"
#include "StraightLine.h"
#include <math.h>


StraightLine::StraightLine(HWND hWnd, int window_width, int window_height)
{
	hWindow = hWnd; // save away window handle
	this->window_width = window_width; // save away window width
	this->window_height = window_height; // save away window width

	ball_colour = RGB(255, 0, 0);
	x1 = 20;
	x2 = 800;
	y1 = 100;
	y2 = 400;
	x = x1;
	y = y1;

	m = (double)(y2 - y1) / (x2 - x1); // slope
	c = (double)(x2 * y1 - x1 * y2) / (x2 - x1); // y-intercept

	// Pen and brush for travelling ball
	ball_pen = CreatePen(PS_SOLID, 1, RGB(0, 0, 0));
	ball_brush = CreateSolidBrush(ball_colour);

	hdc = GetDC(hWindow);

	SelectObject(hdc, ball_pen); // select ball pen colour
	SelectObject(hdc, ball_brush); // select ball brush colour
}

/*
* draws the ball/circle using the apt color
*/
void StraightLine::paint() {
	// draw a dot
	Ellipse(hdc, x, y, x + aWIDTH, y + aHEIGHT);
}

/*
Repeatedly draws ball so as to simulate a continuous motion
*/
void StraightLine::moveInLine() {
	// condition for continuing motion
	while (x + aWIDTH <= window_width) {
		y = (int)ceil(m * x + c);
		paint();

		x += 20;
		// introduce a delay betweeen renderings
		Sleep(50);
	}
}

StraightLine::~StraightLine()
{
	DeleteObject(ball_pen);
	DeleteObject(ball_brush);

	ReleaseDC(hWindow, hdc);
}
